### ISSUE TITLE
Python 3.8 Support

### ISSUE THREAD

**Issue by psifertex** [Vector35/binaryninja-api#1488]
Pyside2 is currently [not compatible](https://bugreports.qt.io/browse/PYSIDE-939) with Python 3.8. In the meantime, dev builds > 1957 have a workaround that will let that version of python still work with the core BN APIs, but UI plugins will not load. 

In the meantime, we recommend using a compatible version of python such as 3.7 or earlier. We'll track any changes here.

**Comment by Ferdi265** [comment]
PySide2 version 5.14 includes a fix compatible with both 3.8 and earlier versions. It's not been released yet, but it's possible to use / compile from git.

I suspect you'd want to wait for it to be released though.

**Comment by Ferdi265** [comment]
I recommend adding a check, so that users who manually updated PySide2 can still use the UI plugins:

something along the lines of:

```python
import sys
import PySide2

def parse_ver(ver):
    # type: (str) -> List[int]
    return [int(n) for n in ver.split('.')]

is_py38 = list(sys.version_info) >= [3, 8]
is_pyside514 = parse_ver(PySide2.__version__) >= [5, 14]

if is_py38 and not is_pyside514:
    print("Python 3.8 is unsupported by PySide2. UI plugins have been disabled.")
```

**Comment by psifertex** [comment]
Since it's relatively close to being fixed properly we'll probably just hold off until the release which sounds like it's not far out.

**Comment by Ferdi265** [comment]
fair point. in the meantime just patching out the check works for me as well :)

**Comment by D0ntPanic** [comment]
Fixed in build 1979. We now use Qt 5.14.0, which was just released and fully supports Python 3.8
